Mathlib/Algebra/Category/ModuleCat/Free.lean

```diff
@@ -33,23 +33,23 @@ linear algebra, module, free
 
 namespace ModuleCat
 
-variable {ι ι' R : Type _}[Ring R] {N P : ModuleCat R} {v : ι → N}
+variable {ι ι' R : Type _} [Ring R] {N P : ModuleCat R} {v : ι → N}
 
-open CategoryTheory
+open CategoryTheory Submodule Set
 
 section LinearIndependent
 
-variable (hv : LinearIndependent R v)  {M : ModuleCat R}
-  {u : ι ⊕ ι' → M}  {f : N ⟶ M} {g : M ⟶ P}
-  (hw : LinearIndependent R (g ∘ u ∘ Sum.inr)) (hu : Function.Injective u)
+variable (hv : LinearIndependent R v) {M : ModuleCat R}
+  {u : ι ⊕ ι' → M} {f : N ⟶ M} {g : M ⟶ P}
+  (hw : LinearIndependent R (g ∘ u ∘ Sum.inr))
   (hm : Mono f) (he : Exact f g) (huv : u ∘ Sum.inl = f ∘ v)
 
-theorem disjoint_span_sum : Disjoint (Submodule.span R (Set.range (u ∘ Sum.inl)))
-    (Submodule.span R (Set.range (u ∘ Sum.inr))) := by
-  rw [huv]
-  refine' Disjoint.mono_left (Submodule.span_mono (Set.range_comp_subset_range _ _)) _
-  rw [← LinearMap.range_coe, (Submodule.span_eq (LinearMap.range f)), (exact_iff _ _).mp he]
-  exact Submodule.ker_range_disjoint (Function.Injective.comp hu Sum.inr_injective) hw
+theorem disjoint_span_sum : Disjoint (span R (range (u ∘ Sum.inl)))
```

Comment:
Can you move this to ` LinearAlgebra/Span`?

Reply:
Ah nevermind, I think I might be missing something.

Reply:
It uses exact sequences so it has to be stated about objects of `ModuleCat`, no?